### DISCUSSION TITLE
Fetching data

### DISCUSSION THREAD

**Discussion by izanoth** [vercel/next.js#69259] [App Router] [answered]
I'm new to react/next.js.
Creating a modest application, I'm trying to make an asynchronous request to an endpoint. The routes work according to the version 13 update (with the app/ directory).
Since I'm new, I'll give you a brief first...
I'm trying to fetch the data with
```
try {
        const res = await fetch('/guest/get');
        if (!res.ok) {
            throw new Error('Failed to fetch guests');
        }
        const data: Guest[] = await res.json();
        setGuests(data);
} catch (error) {
        console.error('Error fetching guests:', error);
}
```
but the server returns 404.
In app/api/guest, I have a get.tsx file (please someone correct me):
```
import { NextRequest, NextResponse } from 'next/server';
import prisma from '../../../lib/prisma';

export async function GET() { 
  try { 
    const guests = await prisma.guest.findMany();
     return NextResponse.json(guests);
  } catch (error) { 
    console.error('Error fetching guests:', error);
    return NextResponse.error();
  } 
} 
``` 
I dynamically read the API reference, but the doubt about the routes persists.
Sorry if it is already solved, i really didn't find out any related.
It cost, tks.

**Comment by icyJoseph** [comment, marked as answer]
Your error, is here:

> In app/api/guest, I have a get.tsx file (please someone correct me):

It should be `app/api/guest/route.tsx`

- https://nextjs.org/docs/app/building-your-application/routing/route-handlers
